lib/Drush/Sql/Sqlmysql.php

```diff
@@ -34,6 +34,12 @@ public function creds($hide_password = TRUE) {
     // server, but not the database.  Connect to the built-in database.
     $parameters['database'] = empty($this->db_spec['database']) ? 'information_schema' : $this->db_spec['database'];
 
+    // If we are creating a new database we don't want to include the database
+    // parameter for the mysql statement.
+    if (isset($this->db_spec['new-database']) && $this->db_spec['new-database']) {
+      unset($parameters['database']);
+    }
+
```

Comment:
Outside of creating or dropping a database I don't see a need to remove the database option. Perhaps changing the key from 'new-database' to 'no-database' or something similar?